pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -2275,6 +2276,17 @@ def test_should_cache(listlike, do_caching):
     )
 
 
+@pytest.mark.parametrize(
+    "listlike",
+    [(deque([pd.Timestamp("2010-06-02 09:30:00")] * 51)),
+        ([pd.Timestamp("2010-06-02 09:30:00")] * 51),
+        (tuple([pd.Timestamp("2010-06-02 09:30:00")] * 51))],
+)
+def test_no_slicing_errors_in_should_cache(listlike):
+    # GH 29403
+    assert tools.should_cache(listlike) is True
```

Comment:
Instead, could you use the example in the original issues as the test? https://github.com/pandas-dev/pandas/issues/29403

Reply:
@mroeschke For this particular bug the example from the original issue could easily stop being indicative if the empirically chosen constants for caching are changed. Let's say if it increased from 50 to 55 or 70 or 100 then the original issue example won't run this code pass, whereas these tests will show that something changed as they would fail. Also they are faster than the original example. And as we definitely know what was the reason of the failure I think it's quite reasonable to use small and fast tests for should_cache() instead.
The original example can be added to ensure the deque works for datetimes. Would be grateful if you could advise to which test file I should add that. Can it be pandas/tests/arrays/test_datetimes.py ?

Reply:
@mroeschke For this particular bug the example from the original issue could easily stop being indicative if the empirically chosen constants for caching are changed. Let's say if it increased from 50 to 55 or 70 or 100 then the original issue example won't run this code pass, whereas these tests will show that something changed as they would fail. Also they are faster than the original example. And as we definitely know what was the reason of the failure I think it's quite reasonable to use small and fast tests for should_cache() instead.
The original example can be added to ensure deque works for timeseries. Would be grateful if you could advise to which test file I should add that. Can it be pandas/tests/arrays/test_datetimes.py ?

Reply:
Good point about the threshold; we can keep the test you have _and_ add the example in the original issue. I think the `pandas/tests/arrays/test_datetimes.py` is a good place for this test.

Reply:
the current location is good for this test

Reply:
@mroeschke I've found a very relevant test `test_to_datetime_cache()` where I included deque in the list of constructors. Actually, with this test, tests I added before for the bug are not much needed any more. What do you think - should I drop them (`test_no_slicing_errors_in_should_cache()` and `test_to_datetime_from_deque()`) now?

Reply:
I'm okay keeping both tests. These tests run relatively quickly.

Reply:
hmm, these do look quite duplicative, though these are hard-coded to the results which is fine.

can you co-located these tests near the above one.